corehq/apps/users/forms.py

```diff
@@ -1436,3 +1477,25 @@ def clean_domains(self):
             domains = EnterprisePermissions.get_domains(self.domain)
             domains += [self.domain]
         return sorted(domains)
+
+    def clean_user_active_status(self):
+        user_active_status = self.cleaned_data['user_active_status']
+
+        if user_active_status == self.ACTIVE:
+            return True
+        if user_active_status == self.INACTIVE:
+            return False
+        return None
+
+    def get_cleaned_data(self):
+        data = self.cleaned_data
+
+        if not data.get('location_id'):
+            # Add (web) user assigned_location_ids so as to
+            # 1) reflect all locations user is assigned to ('All' option)
+            # 2) restrict user access
+            domain_membership = self.couch_user.get_domain_membership(self.domain)
+            if domain_membership.assigned_location_ids:
+                data['web_user_assigned_location_ids'] = domain_membership.assigned_location_ids
```

Comment:
I think this line breaks web user downloads. See https://dimagi-dev.atlassian.net/browse/QA-3723

`domain_membership.assigned_location_ids` is returning a `jsonobject.containers.JsonArray` object, which cannot be pickled at https://github.com/dimagi/commcare-hq/blob/2eea232d71ecd5012d04507143ae7707fc1c0da0/corehq/apps/users/views/mobile/users.py#L1399-L1400